src/node/eviction.h

```diff
@@ -15,55 +15,53 @@
 
 typedef int64_t NodeId;
 
-struct NodeEvictionCandidate {
-    NodeId id;
-    std::chrono::seconds m_connected;
-    std::chrono::microseconds m_min_ping_time;
-    std::chrono::seconds m_last_block_time;
-    std::chrono::seconds m_last_tx_time;
-    bool fRelevantServices;
-    bool m_relay_txs;
-    bool fBloomFilter;
-    uint64_t nKeyedNetGroup;
-    bool prefer_evict;
-    bool m_is_local;
-    Network m_network;
-    bool m_noban;
-    ConnectionType m_conn_type;
+class EvictionManagerImpl;
+
+class EvictionManager
+{
+private:
+    const std::unique_ptr<EvictionManagerImpl> m_impl;
+
+public:
+    explicit EvictionManager();
+    ~EvictionManager();
+
+    void AddCandidate(NodeId id, std::chrono::seconds connected,
+                      uint64_t keyed_net_group, bool prefer_evict,
+                      bool is_local, Network network,
+                      bool noban, ConnectionType conn_type);
+    bool RemoveCandidate(NodeId id);
+
+    /**
+     * Select an inbound peer to evict after filtering out (protecting) peers having
+     * distinct, difficult-to-forge characteristics. The protection logic picks out
+     * fixed numbers of desirable peers per various criteria, followed by (mostly)
+     * ratios of desirable or disadvantaged peers. If any eviction candidates
+     * remain, the selection logic chooses a peer to evict.
+     */
+    [[nodiscard]] std::optional<NodeId> SelectNodeToEvict() const;
+
+    /** A ping-pong round trip has completed successfully. Update minimum ping time. */
+    void UpdateMinPingTime(NodeId id, std::chrono::microseconds ping_time);
+    std::optional<std::chrono::microseconds> GetMinPingTime(NodeId id) const;
```

Comment:
Once the outbound eviction logic has also been moved into EvictionManager, can these three getters be combined into a single `GetEvictionCandidateStats()` or similar, since they'll only be used for reporting stats to the user. If you think that's a good idea, perhaps add that to the function comment for now.